### PR TITLE
Refactor pagination data

### LINKED ISSUE #6581: [API Cleanup] DRY pagination data
### What we should change and why (this is tech debt)

We have a nice standardised format for rendering pagination data in our API endpoints (which was previously reviewed and agreed on), but we have several different copies of the same code for displaying that data, spread around all over the codebase. This needs to be dried up a bit so it's defined clearly in one place.

- https://github.com/openfoodfoundation/openfoodnetwork/blob/76def3a74feb71995b8102da1e07522c57ae1ba3/app/controllers/api/exchange_products_controller.rb#L92-L99
- https://github.com/openfoodfoundation/openfoodnetwork/blob/27586d072da92f22cce13f835722dbb846320d9c/app/controllers/admin/bulk_line_items_controller.rb#L94-L103
- https://github.com/openfoodfoundation/openfoodnetwork/blob/7408c45980b8452c8b0281f6a6318ac590615d89/app/controllers/api/products_controller.rb#L151-L158
- https://github.com/openfoodfoundation/openfoodnetwork/blob/ec4d171993f9a5b2e1ac48a517f8c0806deed87d/app/services/search_orders.rb#L11-L20

### Context

API review.

### PR DESCRIPTION
**Pull request by PauloVilarinho**
#### What? Why?

Closes #6581 

#### What should we test?
Test the pages that use these endpoints:

- **admin/bulk_line_items** (Admin Bulk Order Management page)
- **api/orders** (Admin Orders page)
- **api/exchanges/:exchage_id/products** (Admin Order Cycles pages)
- **api/products/overridable** (Admin Inventory pages)
- **api/products/bulk_products** (Admin Products index page)

#### Release notes
Technical changes

### CODE REVIEW

File: app/controllers/api/exchange_products_controller.rb
Comment (Matt-Yorkley) [inline app/controllers/api/exchange_products_controller.rb:12]: Minor style thing: I think I prefer includes consistently at the top of the class, just under the class declaration and before everything else.

File: app/controllers/api/products_controller.rb
Comment (Matt-Yorkley) [inline app/controllers/api/products_controller.rb:10]: Same here.

File: app/services/search_orders.rb
Comment (Matt-Yorkley) [inline app/services/search_orders.rb:42]: Was there a specific reason for changing this? I have a vague feeling it's coupled to something specific in the Angular code that calls the related endpoint...

File: app/controllers/concerns/pagination_data.rb
Comment (Matt-Yorkley) [inline app/controllers/concerns/pagination_data.rb:6]: I can't quite tell if we really need to keep this `default_page` argument or not. It feels like we could remove it and change `params[:page] || default_page` to `params[:page] || 1` below, and not bother passing in this argument elsewhere in the code? What do you think?

File: app/controllers/api/exchange_products_controller.rb
Comment (Matt-Yorkley) [inline app/controllers/api/exchange_products_controller.rb:88]: These two places where we conditionally render pagination data (here and `Api::OrdersController`) are a bit messy, and we don't even have consistency between the two approaches. I'm wondering if we can tidy this up a bit further and either a) remove the conditional completely, or b) make it nicer and consistent/standardised.

First off I'd rename the above `serializer` variable to `serialized_products`, because that's what it is.

Secondly, I think we could try replacing all of the rest of the lines in this method with something like:
```ruby
render json: { products: serialized_products, pagination: pagination_data(paginated_products) }
```

File: app/controllers/concerns/pagination_data.rb
Comment (Matt-Yorkley) [inline app/controllers/concerns/pagination_data.rb:6]: I think we can remove the `default_per_page` argument from the interface here as well, and replace it with a method like this:

```ruby
private

def default_per_page
  return unless defined? DEFAULT_PER_PAGE

  DEFAULT_PER_PAGE
end
```

As this concern is included in controllers that may or may not define `DEFAULT_PER_PAGE`, we can access it directly from within the concern. So we don't need to pass it as an argument to the `pagination_data` method. What do you think?

Comment (PauloVilarinho) [reply app/controllers/concerns/pagination_data.rb:6]: it would be a great improvement, making the pagination_data function call a lot more cleaner

File: app/services/search_orders.rb
Comment (PauloVilarinho) [reply app/services/search_orders.rb:42]: When the user was sending only the per_page param the application was breaking, but when only the page params were sent it didn't break.

I think this change can be reverted as the paginations calls to this method seems to been called only with both parameters

File: app/controllers/api/exchange_products_controller.rb
Comment (Matt-Yorkley) [reply app/controllers/api/exchange_products_controller.rb]: I think with this guard clause in place, we can also refactor the `Api::OrdersController` index action to just this:

```ruby
def index
  authorize! :admin, Spree::Order

  orders = SearchOrders.new(params, current_api_user).orders

  render json: {
    orders: serialized_orders(orders),
    pagination: pagination_data(orders)
  }
end
```

**Comment by Matt-Yorkley** [comment]
@PauloVilarinho Mind if I push a couple of commits to your branch?

**Comment by PauloVilarinho** [comment]
No problem @Matt-Yorkley

**Comment by PauloVilarinho** [comment]
Do I need to give you access to the repo?

**Comment by Matt-Yorkley** [comment]
Nope, already done. I think there's a setting "allow changes from maintainers" that's enabled by default :+1:

**Review by Matt-Yorkley — APPROVED** [review]
Awesome! :muscle:

I'm pretty sure this will also unblock the Kaminari gem upgrade in issue #6444 :tada:

**Review by andrewpbrett — APPROVED** [review]
Thanks @PauloVilarinho!

**Comment by filipefurtad0** [comment]
Hey @PauloVilarinho ,

I tested the pages you mentioned (before and) after staging the PR, making sure the individual requests are good, see below:

-  admin/bulk_line_items (Admin Bulk Order Management page)
image

- api/orders (Admin Orders page)
image

- api/exchanges/:exchage_id/products (Admin Order Cycles pages)
image

- api/products/overridable (Admin Inventory pages)
image

- api/products/bulk_products (Admin Products index page)
image

All greens. Spotted nothing unusual in the functionality.
Good to go!

**Review by sauloperez — COMMENTED** [review]
Pretty cool @PauloVilarinho ! :clap:

**Pull request merged**
